### PR TITLE
Refactor tests to use pytest features

### PR DESCRIPTION
**Pull request by StrikerRUS**
Use pytest fixtures to parametrize existing tests. It allows to remove duplicated code (mainly, dataset generations), speed up tests, increase code readability. Also it improves test logs readability (see screenshot below), note additional square brackets in a test title.

image

Sorry, this PR looks quite huge for review, but you can hide whitespace changes and leave only meaningful ones (they are all similar across files).

image

### CODE REVIEW

File: lightning/impl/tests/test_dataset.py [hunk 1]
```diff
@@ -10,26 +12,31 @@
...
+    fds = FortranDataset(np.asfortranarray(X))
+    csr_ds = CSRDataset(X_csr)
+    csc_ds = CSCDataset(X_csc)
 
+    return X, X_csr, X_csc, cds, fds, csr_ds, csc_ds
 
-def test_contiguous_get_row():
+
+def test_contiguous_get_row(test_data):
+    X, _, _, cds, _, _, _ = test_data
```
Comment (mblondel) [inline lightning/impl/tests/test_dataset.py:39]: This is not support readable... Maybe `test_data` could be a dictionary instead?

**Review by mblondel — APPROVED** [review]
Huge work, thanks a lot!

File: lightning/impl/tests/test_dataset.py (hunk 1 above)
Comment (mblondel) [reply lightning/impl/tests/test_dataset.py:39]: Yes exactly, just to avoid having to use many underscores.

Comment (StrikerRUS) [reply lightning/impl/tests/test_dataset.py]: Thank you for your review! Addressed in 148635eb40c5d9f918ceb439cc303d68bdc8f923.

**Pull request merged**